ITISFoundation/osparc-simcore
# ♻️🔨  ``api-server``: fixes mypy issues and add type-check step in CI

## What do these changes do?

Fixes mypy errors in ``api-server``  and adds ``type-check`` step in CI


## Related issue/s


- related #4071
- fixes #3551: flaky


## How to test

```cmd
cd services/api-server
make install-dev
make mypy
```
should not show any error
